parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -215,6 +215,54 @@ retrycmd_get_tarball() {
         fi
     done
 }
+retrycmd_acr_access_check() {
+    access_retries=$1; wait_sleep=$2;
+    local acr_url=$3
+    echo "${access_retries} retries for acr access check"
+    sample_image="$acr_url/mcr/hello-world:latest"
+    for i in $(seq 1 $access_retries); do
+        if [ $i -eq $access_retries ]; then
+            return $ERR_ORAS_PULL_NETWORK_ACCESS
+        else
+            timeout 60 oras pull $sample_image --registry-config ${ORAS_REGISTRY_CONFIG_FILE} > $ORAS_OUTPUT 2>&1
```

Comment:
I'd recommend we just store the output of the pull command in a var rather than a new file, as it stands it also looks like that file would be persisted which we don't want